aws/stack/register_middleware.go

```diff
@@ -0,0 +1,54 @@
+package stack
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+// AwsRetryMiddlewareConfig interface for retry middleware config
+type AwsRetryMiddlewareConfig interface {
+	GetRetryer() retry.Retryer
+}
+
+// AddRetryMiddlewares adds retry middleware to operation middleware stack
+func AddRetryMiddlewares(stack *middleware.Stack, cfg AwsRetryMiddlewareConfig,
+	optsFn ...func(*retry.AttemptMiddleware, *retry.MetricsHeaderMiddleware)) {
+	attemptMiddleware := retry.NewAttemptMiddleware(cfg.GetRetryer(), smithyhttp.RequestCloner)
+	metricsHeaderMiddleware := retry.MetricsHeaderMiddleware{}
+	for _, fn := range optsFn {
+		fn(&attemptMiddleware, &metricsHeaderMiddleware)
+	}
+
+	stack.Finalize.Add(attemptMiddleware, middleware.After)
+	stack.Finalize.Add(metricsHeaderMiddleware, middleware.After)
+}
+
+// AddUnsignedPayloadMiddlewares adds UnsignedPayloadMiddleware to the operation middleware stack
+func AddUnsignedPayloadMiddlewares(stack *middleware.Stack, optFns ...func(*v4.UnsignedPayloadMiddleware)) {
+	unsignedPayloadMiddleware := v4.UnsignedPayloadMiddleware{}
+	for _, fn := range optFns {
+		fn(&unsignedPayloadMiddleware)
+	}
+
+	stack.Finalize.Add(&unsignedPayloadMiddleware, middleware.After)
+}
+
+// HTTPSignerMiddlewareConfig interface for HTTP signer middleware config
+type HTTPSignerMiddlewareConfig interface {
+	GetSigner() v4.HTTPSigner
+}
+
+// AddHTTPSignerMiddlewares adds HTTP signer middleware's to operation stack
+func AddHTTPSignerMiddlewares(stack *middleware.Stack, cfg HTTPSignerMiddlewareConfig,
+	optFns ...func( *v4.ComputePayloadSHA256Middleware, *v4.SignHTTPRequestMiddleware)) {
+	computePayloadSHA256Middleware := v4.ComputePayloadSHA256Middleware{}
```

Comment:
Probably want this functional option to be a structure that contains that contains pointers members to these values. That way additional middleware could be added to this helper.